### ISSUE TITLE
Keep organization for the commutative button

### ISSUE THREAD

**Issue by catherinecarter** [phetsims/number-pairs#23]
When in ten-frame organization and a user presses the commutative property button, it would be great (if it's possible) to have the ten-frame organization remain organized if the only action the user has taken is to press the commutative property button.

**Comment by catherinecarter** [comment]
Update: however each side is organized, whether in ten-frame organization or not, when the commutative property button is pressed, mimic and swap left to right addend organization.

**Comment by marlitas** [comment]
This has been addressed in the above commit. Over to @catherinecarter to review.

**Comment by catherinecarter** [comment]
Looks amazing! The only thing I noticed is that the structure isn't maintained for the beads. They currently organize as a clump of beads all together when the commutative property button is pressed. 

All the other representations look great.

**Comment by catherinecarter** [comment]
I'd also like to request a short animation to show the addend objects switching sides. It's difficult to follow and figure out what's happening when the button is pressed since the swap is currently instantaneous.

**Comment by marlitas** [comment]
@catherinecarter can you review the behavior on the beads now?

> I'd also like to request a short animation to show the addend objects switching sides.

I'm assuming this is for the location based objects. I would not recommend animating the beads.

**Comment by marlitas** [comment]
@catherinecarter I started to work on animating the location objects across and this is actually a way bigger lift after I remembered that the objects aren't technically switching sides in the code, so any animation is not going to look clean the way you're imagining... 

Here's a patch that I can use to show you more once we can chat in person.

<details>

```diff
Subject: [PATCH] swap addends animation
---
Index: js/common/model/NumberPairsModel.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/js/common/model/NumberPairsModel.ts b/js/common/model/NumberPairsModel.ts
--- a/js/common/model/NumberPairsModel.ts	(revision bfa847fc2005d5c4ece49e8ed6907cac3a486682)
+++ b/js/common/model/NumberPairsModel.ts	(date 1735071935611)
@@ -73,7 +73,8 @@
   public readonly leftAddendLabelPlacementProperty: Property<leftAddendLabelPlacement>;
 
   private dropAnimation: Animation | null = null;
-  private tenFrameAnimation: Animation | null = null;
+  private attributeAnimation: Animation | null = null;
+  private locationAnimation: Animation | null = null;
 
   public readonly groupSelectLocationObjectsModel: GroupSelectModel<CountingObject>;
   public readonly groupSelectBeadsModel: GroupSelectModel<CountingObject>;
@@ -396,7 +397,7 @@
       new Vector2( position.x + xTranslation, position.y ) );
     const newLeftLocationPositions = rightLocationPositions.map( position =>
       new Vector2( position.x - xTranslation, position.y ) );
-    this.setLocationPositions( newLeftLocationPositions, newRightLocationPositions );
+    this.setLocationPositions( newLeftLocationPositions, newRightLocationPositions, true );
 
     // Bead positions should be a translation across the separator.
     const updatedSeparatorPosition = NumberPairsModel.calculateBeadSeparatorXPosition( this.leftAddendProperty.value );
@@ -410,44 +411,89 @@
     assert && assert( this.rightAddendCountingObjectsProperty.value.length === this.rightAddendProperty.value, 'Addend array length and value should match' );
   }
 
+  /**
+   * Returns animation targets based on the provided position Properties and target positions.
+   * @param properties
+   * @param targetPositions
+   */
+  private getAnimationTargets( positionProperties: Property<Vector2>[], targetPositions: Vector2[] ): AnimationTarget[] {
+    return positionProperties.map( ( positionProperty, index ) => {
+      return {
+        property: positionProperty,
+        to: targetPositions[ index ]
+      };
+    } );
+  }
+
   /**
    * Set the location positions of the counting objects based on the provided left and right location positions.
    * @param leftLocationPositions
    * @param rightLocationPositions
+   * @param animate - whether to animate the movement of the counting objects.
    */
-  public setLocationPositions( leftLocationPositions: Vector2[], rightLocationPositions: Vector2[] ): void {
+  public setLocationPositions( leftLocationPositions: Vector2[], rightLocationPositions: Vector2[], animate = false ): void {
     const leftAddendObjects = this.leftAddendCountingObjectsProperty.value;
     const rightAddendObjects = this.rightAddendCountingObjectsProperty.value;
 
     assert && assert( leftAddendObjects.length === leftLocationPositions.length, 'leftAddendObjects should be the same length as the rightLocationPositions now that they have been swapped.' );
     assert && assert( rightAddendObjects.length === rightLocationPositions.length, 'rightAddendObjects should be the same length as the leftLocationPositions now that they have been swapped.' );
 
-    leftAddendObjects.forEach( ( countingObject, index ) => {
-      countingObject.locationPositionProperty.value = leftLocationPositions[ index ];
-    } );
-    rightAddendObjects.forEach( ( countingObject, index ) => {
-      countingObject.locationPositionProperty.value = rightLocationPositions[ index ];
-    } );
+    if ( animate ) {
+      const animationTargets = [ ...this.getAnimationTargets( leftAddendObjects.map( countingObject => countingObject.locationPositionProperty ), leftLocationPositions ),
+                                 ...this.getAnimationTargets( rightAddendObjects.map( countingObject => countingObject.locationPositionProperty ), rightLocationPositions ) ];
+      this.locationAnimation = new Animation( {
+        duration: 0.4,
+        targets: animationTargets
+      } );
+      this.locationAnimation.endedEmitter.addListener( () => {
+        this.locationAnimation = null;
+      } );
+      this.locationAnimation.start();
+    }
+    else {
+      leftAddendObjects.forEach( ( countingObject, index ) => {
+        countingObject.locationPositionProperty.value = leftLocationPositions[ index ];
+      } );
+      rightAddendObjects.forEach( ( countingObject, index ) => {
+        countingObject.locationPositionProperty.value = rightLocationPositions[ index ];
+      } );
+    }
   }
 
   /**
    * Set the attribute positions of the counting objects based on the provided left and right addend positions.
    * @param leftAttributePositions
    * @param rightAttributePositions
+   * @param animate - whether to animate the movement of the counting objects.
    */
-  public setAttributePositions( leftAttributePositions: Vector2[], rightAttributePositions: Vector2[] ): void {
+  public setAttributePositions( leftAttributePositions: Vector2[], rightAttributePositions: Vector2[], animate = false ): void {
     const leftAddendObjects = this.leftAddendCountingObjectsProperty.value;
     const rightAddendObjects = this.rightAddendCountingObjectsProperty.value;
 
     assert && assert( leftAddendObjects.length === leftAttributePositions.length, 'leftAddendObjects should be the same length as the rightAttributePositions now that they have been swapped.' );
     assert && assert( rightAddendObjects.length === rightAttributePositions.length, 'rightAddendObjects should be the same length as the leftAttributePositions now that they have been swapped.' );
 
-    leftAddendObjects.forEach( ( countingObject, index ) => {
-      countingObject.attributePositionProperty.value = leftAttributePositions[ index ];
-    } );
-    rightAddendObjects.forEach( ( countingObject, index ) => {
-      countingObject.attributePositionProperty.value = rightAttributePositions[ index ];
-    } );
+    if ( animate ) {
+      const animationTargets = [ ...this.getAnimationTargets( leftAddendObjects.map( countingObject => countingObject.attributePositionProperty ), leftAttributePositions ),
+        ...this.getAnimationTargets( rightAddendObjects.map( countingObject => countingObject.attributePositionProperty ), rightAttributePositions ) ];
+
+      this.attributeAnimation = new Animation( {
+        duration: 0.4,
+        targets: animationTargets
+      } );
+      this.attributeAnimation.endedEmitter.addListener( () => {
+        this.attributeAnimation = null;
+      } );
+      this.attributeAnimation.start();
+    }
+    else {
+      leftAddendObjects.forEach( ( countingObject, index ) => {
+        countingObject.attributePositionProperty.value = leftAttributePositions[ index ];
+      } );
+      rightAddendObjects.forEach( ( countingObject, index ) => {
+        countingObject.attributePositionProperty.value = rightAttributePositions[ index ];
+      } );
+    }
   }
 
   /**
@@ -520,52 +566,22 @@
   public organizeIntoTenFrame( tenFrameBounds: Bounds2[] ): void {
     assert && assert( tenFrameBounds.length === 1 || tenFrameBounds.length === 2, 'Ten frame bounds must be an array of length 1 or 2.' );
     const leftAddendObjects = this.leftAddendCountingObjectsProperty.value;
-    const rightAddendObjects = this.rightAddendCountingObjectsProperty.value;
 
-    // If we are only provided one ten frame bound, we will organize the objects within that single ten frame bounds.
+    // If we are only provided one ten frame bound, we are in the unified counting area where Counting Objects are split by attribute.
+    // If we have two ten frame bounds, we are in a split counting area where Counting Objects are split by location.
     let leftGridCoordinates: Vector2[];
     let rightGridCoordinates: Vector2[];
     if ( tenFrameBounds.length === 1 ) {
       const gridCoordinates = this.getGridCoordinates( tenFrameBounds[ 0 ], 0, 0 );
       leftGridCoordinates = gridCoordinates.slice( 0, leftAddendObjects.length );
       rightGridCoordinates = gridCoordinates.slice( leftAddendObjects.length );
+      this.setAttributePositions( leftGridCoordinates, rightGridCoordinates, true );
     }
-    else {
+    else if ( tenFrameBounds.length === 2 ) {
       leftGridCoordinates = this.getGridCoordinates( tenFrameBounds[ 0 ], 20, 50 );
       rightGridCoordinates = this.getGridCoordinates( tenFrameBounds[ 1 ], 50, 20 );
+      this.setLocationPositions( leftGridCoordinates, rightGridCoordinates, true );
     }
-
-
-    const tenFrameAnimationTargets: AnimationTarget[] = [];
-    leftAddendObjects.forEach( ( countingObject, index ) => {
-      tenFrameAnimationTargets.push( {
-        property: countingObject.attributePositionProperty,
-        to: leftGridCoordinates[ index ]
-      } );
-      tenFrameAnimationTargets.push( {
-        property: countingObject.locationPositionProperty,
-        to: leftGridCoordinates[ index ]
-      } );
-    } );
-    rightAddendObjects.forEach( ( countingObject, index ) => {
-      tenFrameAnimationTargets.push( {
-        property: countingObject.attributePositionProperty,
-        to: rightGridCoordinates[ index ]
-      } );
-      tenFrameAnimationTargets.push( {
-        property: countingObject.locationPositionProperty,
-        to: rightGridCoordinates[ index ]
-      } );
-    } );
-
-    this.tenFrameAnimation = new Animation( {
-      duration: 0.4,
-      targets: tenFrameAnimationTargets
-    } );
-    this.tenFrameAnimation.endedEmitter.addListener( () => {
-      this.tenFrameAnimation = null;
-    } );
-    this.tenFrameAnimation.start();
 
     assert && assert( this.leftAddendCountingObjectsProperty.value.length === this.leftAddendProperty.value, 'Addend array length and value should match' );
     assert && assert( this.rightAddendCountingObjectsProperty.value.length === this.rightAddendProperty.value, 'Addend array length and value should match' );
@@ -609,7 +625,8 @@
 
     // Stop any animations that may be in progress.
     this.dropAnimation?.stop();
-    this.tenFrameAnimation?.stop();
+    this.attributeAnimation?.stop();
+    this.locationAnimation?.stop();
 
     this.representationTypeProperty.reset();
     this.leftAddendVisibleProperty.reset();
```

</details>

**Comment by catherinecarter** [comment]
@marlitas and I discussed and decided to try fading the objects out and back into their respective locations after pressing the swap button to avoid the issue described above.

Testing fade.

**Comment by marlitas** [comment]
The fade has been added in. This is ready for @catherinecarter's review.

**Comment by catherinecarter** [comment]
The fade looks good for the location-based objects. To verify, this will be added in for the beads as well?

Two things I noticed that could use improvement. 
1. If we center the location-based objects, when organized using the organize button, the swap might look less jarring:  <img width="1016" alt="image" src="https://github.com/user-attachments/assets/b6491ad1-5b40-4ede-a2c1-87241497ea29" />

2. This comment might inspire table-flipping, but I'm wondering if mirroring the objects rather than swapping their exact arrangement might help alleviate this scenario: 
<img width="1010" alt="image" src="https://github.com/user-attachments/assets/f2176cb3-f63e-49d4-9e3e-420bd78db3f8" /> to  <img width="1020" alt="image" src="https://github.com/user-attachments/assets/afb9f543-ccba-482c-9bb0-f43b9ab694a1" /> where the lower left image on the right side is now covering the eye toggle on the left after swapping. How much of a lift would it be to mirror rather than swap?

**Comment by marlitas** [comment]
@catherinecarter and I chatted this through. I think it will be easy to implement. Here's the suggested math:

    TODO Mirror:
    1. Find the distance from center of object to center of counting area
    2. Multiply distance by 2
    3. Add or subtract (depending on direction) from object's x position

**Comment by catherinecarter** [comment]
Pinging this to avoid this getting lost: 
- Fade is effective. We'd like this on the beads and kittens as well.

**Comment by marlitas** [comment]
> Fade is effective. We'd like this on the beads and kittens as well.

Please don't ask me to do this on the beads... they are such a headache and this is just going to add on another layer that I already foresee too many days of work on. @catherinecarter

**Comment by catherinecarter** [comment]
When sifting through the issues for alt-input, I noticed when I push the commutative button, there's a object appearing at a random location just before the fade. This happens for both alt-input and mouse input. See video below. Putting this here because it's related to the commutative button. Should I open a new issue and move it there?

https://github.com/user-attachments/assets/cf911a5b-8cd2-4c40-aeb5-f929451501bc

**Comment by marlitas** [comment]
This is ready for @catherinecarter's review. The bug you reported in the previous comment has also been addressed. 
Keep this open, I will send over to @pixelzoom once design review is done.

**Comment by catherinecarter** [comment]
Can we reduce the fade time? The mirroring is nice, I like it better than the swap, but it gets lost since the fade is long.

I don't have a suggested number since I don't know what it is currently, but maybe cut in 1/2 as a start?

**Comment by marlitas** [comment]
Cut the animation in half. Should be quicker now.

**Comment by catherinecarter** [comment]
Looks good now. Speed is nice - easier to track the positions of the objects.

**Comment by pixelzoom** [comment]
@marlitas Can you summarize what I need to code review? I've lost track, and there are a lot of comments in this issue.

**Comment by marlitas** [comment]
@pixelzoom yes here is a list of specific items that were affected:
- `swapAddends()` in `NumberPairsModel`
- `setLocationPositions()` in `NumberPairsModel`
- `shiftOverlappingBeadPositions()` in `BeadManager`

Mostly anything that may be affected from `swapAddends` will be good to get an eye on.

**Comment by pixelzoom** [comment]
I reviewed the methods identified in https://github.com/phetsims/number-pairs/issues/23#issuecomment-2657266683. They are nicely documented, and I don't see any obvious problems.  Playing with the sim, I don't see any obvious bugs.  Nicely done - back to @marlitas for next steps.

**Comment by marlitas** [comment]
Great. I think we can close. Thanks everyone!
